### PR TITLE
Mutation/Aggregations Merge and Decimal Fix

### PR DESCRIPTION
**Pull request by JZFLei**
My PR is a:
🐛 Bug fix
💅 Enhancement

Main update on the:
Documentation
Plugin GraphQL

Merging Mutations (#1650) and Aggregations.

Fixed Mongoose Hook to use Decimals (2 decimal places) and Floats (20 decimal places).

### CODE REVIEW

File: packages/strapi-hook-mongoose/lib/utils/index.js [hunk 1]
```diff
@@ -5,7 +5,8 @@
  */
 
 module.exports = mongoose => {
-  require('mongoose-float').loadType(mongoose);
+  var Decimal = require('mongoose-float').loadType(mongoose, 2);
```
Comment (z-ax) [inline packages/strapi-hook-mongoose/lib/utils/index.js]: Why `var` while the most uses `const` etc?

Comment (TokenChingy) [reply packages/strapi-hook-mongoose/lib/utils/index.js]: Due to the 'use strict' tag which enforces ES5 therefore `const` and `let` cannot be used.

Comment (kamalbennani) [reply packages/strapi-hook-mongoose/lib/utils/index.js]: Really? we do use const/let everywhere in the app.

Comment (TokenChingy) [reply packages/strapi-hook-mongoose/lib/utils/index.js]: I don’t mind using them but I’m working off whatever code was there initially.

**Comment by TokenChingy** [comment]
@lauriejim can you review or assign someone to review to get this merged?

**Comment by lauriejim** [comment]
Can you try a clean install of your branch on your computer and be sure all is ok please.
I think you will have some issues. Can you confirm if all is OK.

**Comment by TokenChingy** [comment]
Will have a look in the morning.

**Comment by NickBolles** [comment]
@TokenChingy @lauriejim Why in the world was all of the logic to create resolver duplicated for mutations and queries? 90% of the code to create a resolver is the same between mutation and queries, why not break those parts into their own functions instead of straight up duplicating it?

**Comment by Aurelsicoko** [comment]
@NickBolles In this world, the more project you develop, the more you know that duplicate code isn't a bad thing. I really prefer duplicated code which is maintainable and readable than having one function for handling mutation and resolver with a lot of conditions and very complicated to read.

There is no performance issue by doing it like this and it's easier to maintain so we will continue to split both logic into two different files, even if there are a lot of similarities.

**Comment by NickBolles** [comment]
@Aurelsicoko I work at a giant software company where it is common to copy the code and modify it a bit. And I can say with 100% certainty that duplicate code is terrible. Because then someone like me comes and wants to make a change to queries and has to make it in two places and then gives up because the code is too convoluted. 

Honestly your going to both extremes. There's a middle ground. It's called function programming. Or at least a sprinkle of it. Take the common parts, break them out into pure functions that take inputs and give predictable outputs. Unit test them and use them everywhere that functionality is used. This makes it so sure you have two different code paths, in this case for queries and resolvers, but they are both composing a result based off of different inputs to common functions. 

Strapi does just about zero of this. Even though stuff like looping over all of the models, or finding a model based on if it's in a plugin or not. That makes the code very tied together and not encapsulated at all. 

I really think it is something that strapi should work on improving, and I actually got a long way through a PR that had about 10 utility functions that would cut out a few hundred lines of code throughout the code base, and make it a descriptive, one liner. 

Check out my changes (and a lot of noise, because apparently I just did a commit all with whitespace changes...) In this branch for a small example of what I'm talking about.
https://github.com/strapi/strapi/compare/master...NickBolles:njb/attribute-permissions

**Comment by Aurelsicoko** [comment]
I agree with you on most of points. However, in this specific case, I still think it's a great idea to split into both files. However, as I said, there are a lot of similarities and we can try to create pure functions with predictable results. We have the same vision of things, don't worry... I was just trying to explain that I prefer maintainable and readable code than one function for handling two things.

We can still have two files with a lot of pure functions which are called and shared across the `Query.js` and `Mutation.js` files 👍

We know that the current codebase isn't optimal at all. We have bootstrapped a lot of things to be able to move fast and create this community (and also secure fundraising). We are doing our best to change that and there will have a lot of changes in the next months to improve the code quality.

File: packages/strapi-hook-mongoose/lib/utils/index.js [hunk 2]
```diff
@@ -22,9 +23,9 @@ module.exports = mongoose => {
         case 'timestamp':
           return Date;
         case 'decimal':
-          return 'Float';
+          return Decimal;
         case 'float':
-          return mongoose.Schema.Types.Decimal128;
+          return Float;
```
Comment (henrych4) [inline packages/strapi-hook-mongoose/lib/utils/index.js]: Does it break previous records with field of float type?

**Pull request merged**